### PR TITLE
#7519 feat: PageHeaderCompact component

### PR DESCRIPTION
**Pull request by chriddell**
Resolves wellcometrust/corporate/issues/7519

- adds PageHeaderCompact component

### CODE REVIEW

File: src/components/PageHeaderCompact/PageHeaderCompact.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+
+import Author, { AuthorProps } from 'Author/Author';
+import { ImageElement } from 'Image';
+import RichText from 'RichText';
```
Comment (creido-welly) [inline src/components/PageHeaderCompact/PageHeaderCompact.tsx:7]: Minor: alphabetical order

File: src/components/PageHeaderCompact/PageHeaderCompact.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import TagList, { TagProps } from 'TagList/TagList';
+import SocialShare from 'SocialShare';
+import FormattedDate from 'FormattedDate';
+
+interface AuthorPropsExtend extends AuthorProps {
```
Comment (creido-welly) [inline src/components/PageHeaderCompact/PageHeaderCompact.tsx]: type is preferable to interface as interface can easily inadvertently acquire props from other components, in which case you can change this to 

```type AuthorPropsExtend = AuthorProps &  {```

Example in use for the Card component
https://github.com/wellcometrust/corporate-components/blob/e4812b145f31b2db9fb84db0460778c6beaddf49/src/components/Card/Card.tsx#L18

File: src/components/PageHeaderCompact/PageHeaderCompact.tsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  imageLicence?: string;
+  metaLabel?: string;
+  socialUrl: string;
+  standfirst: string;
+  title: string;
+  topics: TagProps[];
+};
+
+export const PageHeaderCompact = ({
+  authors,
```
Comment (creido-welly) [inline src/components/PageHeaderCompact/PageHeaderCompact.tsx:32]: Minor: alphabetical order

File: src/components/PageHeaderCompact/PageHeaderCompact.tsx (hunk 2 above)
Comment (chriddell) [reply src/components/PageHeaderCompact/PageHeaderCompact.tsx]: ah amazing, I thought had to use an interface in order to extend. will update this

File: src/components/PageHeaderCompact/PageHeaderCompact.tsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  imageCaption,
+  imageCredit,
+  imageLicence,
+  imageSizes,
+  imageSrc,
+  imageSrcSet,
+  metaLabel,
+  title,
+  topics,
+  socialUrl,
```
Comment (creido-welly) [inline src/components/PageHeaderCompact/PageHeaderCompact.tsx:43]: Minor: alphabetical order

File: src/components/PageHeaderCompact/PageHeaderCompact.tsx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+          {imageLicence && (
+            <div
+              className="cc-media__licence"
+              dangerouslySetInnerHTML={{ __html: `Licence: ${imageLicence}` }}
+            />
+          )}
+        </figcaption>
+      </figure>
+    )}
+    <div className="cc-page-header-compact__aside cc-page-header-compact__section cc-page-header-compact__section--sidebar">
```
Comment (creido-welly) [inline src/components/PageHeaderCompact/PageHeaderCompact.tsx:73]: Do you think this could be an `<aside />` element?

Comment (chriddell) [reply src/components/PageHeaderCompact/PageHeaderCompact.tsx:73]: good question, `<aside>` spec says it...

> represents a portion of a document whose content is only indirectly related to the document's main content

so the question is are authors and topic tags directly or indirectly related to the article content? Imo at the very least, author is directly related

File: src/components/PageHeaderCompact/PageHeaderCompact.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        </div>
+      )}
+      {topics && (
+        <div className="cc-page-header-compact__topics">
+          <TagList tags={topics} />
+        </div>
+      )}
+    </div>
+    <div className="cc-page-header-compact__tray cc-page-header-compact__section cc-page-header-compact__section--main">
+      <time dateTime={date}>
```
Comment (creido-welly) [inline src/components/PageHeaderCompact/PageHeaderCompact.tsx:96]: Just looking across this library, I think the `<time>` element could be incorporated into the `<FormattedDate>` component - only ResultsItem does wrap FD with `<time>` and it prob should. Could be a future PR tho.

Can you think of any reason we wouldn't want to wrap it?

Comment (chriddell) [reply src/components/PageHeaderCompact/PageHeaderCompact.tsx:96]: I have a feeling `<time>` could be the element that causes iOS (and potentially Android/other mobile OS') to allow the user to add the date to their calendar... Not sure if this should be a reason to stop us doing it, though, but could create a style issue?

File: src/components/PageHeaderCompact/PageHeaderCompact.tsx (hunk 5 above)
Comment (creido-welly) [reply src/components/PageHeaderCompact/PageHeaderCompact.tsx:73]: True

File: src/components/PageHeaderCompact/PageHeaderCompact.tsx (hunk 6 above)
Comment (creido-welly) [reply src/components/PageHeaderCompact/PageHeaderCompact.tsx:96]: Ticket created for this https://github.com/wellcometrust/corporate/issues/7529

We can take a view further down the line whether to do it or not

**Review by creido-welly — APPROVED** [review]

**Pull request merged**